diff --git a/package/boot/uboot-envtools/files/qualcommax b/package/boot/uboot-envtools/files/qualcommax
index 0c6b9e60b..e7c7992e9 100644
--- a/package/boot/uboot-envtools/files/qualcommax
+++ b/package/boot/uboot-envtools/files/qualcommax
@@ -48,6 +48,8 @@ qnap,301w)
 	[ -n "$idx" ] && \
 		ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x20000" "0x20000" "1"
 	;;
+jdcloud,ax6600|\
+jdcloud,ax1800-pro|\
 redmi,ax5-jdcloud)
 	mmcpart="$(find_mmc_part 0:APPSBLENV)"
 	[ -n "$mmcpart" ] && \
diff --git a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-re-ss-01.dts b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-ax1800-pro.dts
similarity index 94%
rename from target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-re-ss-01.dts
rename to target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-ax1800-pro.dts
index 3a4ec3caa..cd219f4c3 100644
--- a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-re-ss-01.dts
+++ b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6000-ax1800-pro.dts
@@ -10,15 +10,15 @@
 #include <dt-bindings/input/input.h>
 
 / {
-	model = "JDCloud RE-SS-01";
-	compatible = "jdcloud,re-ss-01", "qcom,ipq6018";
+	model = "JDCloud AX1800 Pro";
+	compatible = "jdcloud,ax1800-pro", "qcom,ipq6018";
 
 	aliases {
 		serial0 = &blsp1_uart3;
-		led-boot = &led_status_green;
+		led-boot = &led_status_red;
 		led-failsafe = &led_status_red;
-		led-running = &led_status_blue;
-		led-upgrade = &led_status_red;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_green;
 
 		ethernet1 = "/soc/dp2";
 		ethernet2 = "/soc/dp3";
diff --git a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-re-cs-02.dts b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-ax6600.dts
similarity index 96%
rename from target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-re-cs-02.dts
rename to target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-ax6600.dts
index a7ec1be5f..17037a44f 100644
--- a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-re-cs-02.dts
+++ b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6010-ax6600.dts
@@ -10,16 +10,16 @@
 #include <dt-bindings/input/input.h>
 
 / {
-	model = "JDCloud RE-CS-02";
-	compatible = "jdcloud,re-cs-02", "qcom,ipq6018";
+	model = "JDCloud AX6600";
+	compatible = "jdcloud,ax6600", "qcom,ipq6018";
 
 	aliases {
 		serial0 = &blsp1_uart3;
 		serial1 = &blsp1_uart6;
-		led-boot = &led_status_green;
+		led-boot = &led_status_red;
 		led-failsafe = &led_status_red;
-		led-running = &led_status_blue;
-		led-upgrade = &led_status_red;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_green;
 
 		ethernet0 = "/soc/dp1";
 		ethernet1 = "/soc/dp2";
diff --git a/target/linux/qualcommax/image/ipq60xx.mk b/target/linux/qualcommax/image/ipq60xx.mk
index ef05a150b..4b203b733 100644
--- a/target/linux/qualcommax/image/ipq60xx.mk
+++ b/target/linux/qualcommax/image/ipq60xx.mk
@@ -18,9 +18,8 @@ define Device/UbiFit
 endef
 
 define Device/EmmcImage
-	IMAGES := factory.bin recovery.bin sysupgrade.bin
-	IMAGE/factory.bin := append-kernel | pad-to 12288k | append-rootfs | append-metadata
-	IMAGE/recovery.bin := append-kernel | pad-to 6144k | append-rootfs | append-metadata
+	IMAGES := factory.bin sysupgrade.bin
+	IMAGE/factory.bin := append-kernel | pad-to 6144k | append-rootfs | append-metadata
 endef
 
 define Device/cmiot_ax18
@@ -61,7 +60,7 @@ define Device/glinet_gl-axt1800
 endef
 TARGET_DEVICES += glinet_gl-axt1800
 
-define Device/jdcloud_re-cs-02
+define Device/jdcloud_ax6600
 	$(call Device/FitImage)
 	$(call Device/EmmcImage)
 	DEVICE_VENDOR := JDCloud
@@ -72,9 +71,9 @@ define Device/jdcloud_re-cs-02
 	DEVICE_DTS_CONFIG := config@cp03-c3
 	DEVICE_PACKAGES := ipq-wifi-jdcloud_ax6600 kmod-ath11k-pci ath11k-firmware-qcn9074
 endef
-TARGET_DEVICES += jdcloud_re-cs-02
+TARGET_DEVICES += jdcloud_ax6600
 
-define Device/jdcloud_re-ss-01
+define Device/jdcloud_ax1800-pro
 	$(call Device/FitImage)
 	$(call Device/EmmcImage)
 	DEVICE_VENDOR := JDCloud
@@ -85,7 +84,7 @@ define Device/jdcloud_re-ss-01
 	DEVICE_DTS_CONFIG := config@cp03-c2
 	DEVICE_PACKAGES := ipq-wifi-jdcloud_ax1800pro
 endef
-TARGET_DEVICES += jdcloud_re-ss-01
+TARGET_DEVICES += jdcloud_ax1800-pro
 
 define Device/linksys_mr7350
 	$(call Device/FitImage)
diff --git a/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network b/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network
index fc123c9a1..4266cd42a 100644
--- a/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network
+++ b/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network
@@ -12,14 +12,14 @@ ipq60xx_setup_interfaces()
 
 	case "$board" in
 	cmiot,ax18|\
-	jdcloud,re-ss-01|\
+	jdcloud,ax1800-pro|\
 	qihoo,360v6|\
 	redmi,ax5-*|\
 	xiaomi,rm1800)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
 		;;
 	glinet,gl-ax1800|\
-	jdcloud,re-cs-02|\
+	jdcloud,ax6600|\
 	linksys,mr7350)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4" "wan"
 		;;
diff --git a/target/linux/qualcommax/ipq60xx/base-files/etc/hotplug.d/firmware/11-ath11k-caldata b/target/linux/qualcommax/ipq60xx/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
index b82946cdc..b4c880a4b 100644
--- a/target/linux/qualcommax/ipq60xx/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
+++ b/target/linux/qualcommax/ipq60xx/base-files/etc/hotplug.d/firmware/11-ath11k-caldata
@@ -16,8 +16,8 @@ case "$FIRMWARE" in
 	xiaomi,rm1800)
 		caldata_extract "0:art" 0x1000 0x10000
 		;;
-	jdcloud,re-cs-02|\
-	jdcloud,re-ss-01|\
+	jdcloud,ax6600|\
+	jdcloud,ax1800-pro|\
 	redmi,ax5-jdcloud)
 		caldata_extract_mmc "0:ART" 0x1000 0x10000
 		;;
@@ -25,7 +25,7 @@ case "$FIRMWARE" in
 	;;
 "ath11k/QCN9074/hw1.0/cal-pci-0000:01:00.0.bin")
 	case "$board" in
-	jdcloud,re-cs-02)
+	jdcloud,ax6600)
 		caldata_extract_mmc "0:ART" 0x26800 0x20000
 		;;
 	esac
diff --git a/target/linux/qualcommax/ipq60xx/base-files/lib/preinit/81_fix_eeprom b/target/linux/qualcommax/ipq60xx/base-files/lib/preinit/81_fix_eeprom
index 34b91f4df..0ce0a4860 100644
--- a/target/linux/qualcommax/ipq60xx/base-files/lib/preinit/81_fix_eeprom
+++ b/target/linux/qualcommax/ipq60xx/base-files/lib/preinit/81_fix_eeprom
@@ -2,8 +2,8 @@
 
 preinit_fix_eeprom() {
 	case $(board_name) in
-	jdcloud,re-cs-02|\
-	jdcloud,re-ss-01|\
+	jdcloud,ax6600|\
+	jdcloud,ax1800-pro|\
 	redmi,ax5-jdcloud)
 		mmc_part=$(find_mmc_part 0:ART)
 		FIRMWARE=""ath11k/IPQ6018/hw1.0/cal-ahb-c000000.wifi.bin""
diff --git a/target/linux/qualcommax/ipq60xx/base-files/lib/upgrade/platform.sh b/target/linux/qualcommax/ipq60xx/base-files/lib/upgrade/platform.sh
index e199c33c1..3960287cb 100644
--- a/target/linux/qualcommax/ipq60xx/base-files/lib/upgrade/platform.sh
+++ b/target/linux/qualcommax/ipq60xx/base-files/lib/upgrade/platform.sh
@@ -17,8 +17,8 @@ platform_do_upgrade() {
 	xiaomi,rm1800)
 		nand_do_upgrade "$1"
 		;;
-	jdcloud,re-cs-02|\
-	jdcloud,re-ss-01|\
+	jdcloud,ax6600|\
+	jdcloud,ax1800-pro|\
 	redmi,ax5-jdcloud)
 		kernelname="0:HLOS"
 		rootfsname="rootfs"
