diff --git a/package/network/config/wifi-scripts/files/lib/wifi/mac80211.sh b/package/network/config/wifi-scripts/files/lib/wifi/mac80211.sh
index e24a2a634e..4a3d6455b5 100644
--- a/package/network/config/wifi-scripts/files/lib/wifi/mac80211.sh
+++ b/package/network/config/wifi-scripts/files/lib/wifi/mac80211.sh
@@ -203,7 +203,7 @@ detect_mac80211() {
 			set wireless.${name}.channel=${channel}
 			set wireless.${name}.band=${mode_band}
 			set wireless.${name}.htmode=$htmode
-			set wireless.${name}.disabled=1
+			set wireless.${name}.disabled=0
 
 			set wireless.default_${name}=wifi-iface
 			set wireless.default_${name}.device=${name}
diff --git a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-jdc-ax1800-pro.dts b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-jdc-ax1800-pro.dts
index 5c743d03be..1563b46d45 100644
--- a/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-jdc-ax1800-pro.dts
+++ b/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq6018-jdc-ax1800-pro.dts
@@ -20,7 +20,6 @@
 #include "ipq6018-ess.dtsi"
 #include "ipq6018-nss.dtsi"
 
-
 #include <dt-bindings/input/input.h>
 #include <dt-bindings/gpio/gpio.h>
 #include <dt-bindings/leds/common.h>
@@ -98,7 +97,7 @@
 		mux {
 			pins = "gpio38","gpio39","gpio40","gpio41";
 			function = "blsp0_spi";
-			drive-strength = <0x08>;
+			drive-strength = <8>;
 			bias-pull-down;
 		};
 	};
@@ -125,21 +124,21 @@
 		mux_0 {
 			pins = "gpio64";
 			function = "mdc";
-			drive-strength = <0x08>;
+			drive-strength = <8>;
 			bias-pull-up;
 		};
 
 		mux_1 {
 			pins = "gpio65";
 			function = "mdio";
-			drive-strength = <0x08>;
+			drive-strength = <8>;
 			bias-pull-up;
 		};
 
 		mux_2 {
 			pins = "gpio75";
 			function = "gpio";
-		bias-pull-up;
+			bias-pull-up;
 		};
 	};
 };
@@ -192,7 +191,6 @@
 	status = "okay";
 };
 
-
 &ssphy_0 {
 	status = "okay";
 };
@@ -283,39 +281,43 @@
 };
 
 &wifi {
-	status = "okay";
 	qcom,ath11k-calibration-variant = "JDC-AX1800-Pro";
 	qcom,ath11k-fw-memory-mode = <1>;
+	status = "okay";
 };
 
 
 &dp2 {
 	phy-handle = <&qca8075_1>;
+	label = "lan1";
 	status = "okay";
 };
 
 &dp3 {
 	phy-handle = <&qca8075_2>;
+	label = "lan2";
 	status = "okay";
 };
 
 &dp4 {
 	phy-handle = <&qca8075_3>;
+	label = "lan3";
 	status = "okay";
 };
 
 &dp5 {
 	phy-handle = <&qca8075_4>;
 	phy-mode = "psgmii";
+	label = "wan";
 	status = "okay";
 };
 
 &sdhc_1 {
-	status = "okay";
-
-	/delete-property/ mmc-hs400-1_8v;
-	mmc-hs200-1_8v;
+	bus-width = <8>;
 	mmc-ddr-1_8v;
+	mmc-hs200-1_8v;
+	non-removable;
+	status = "okay";
 };
 
 &CPU0 {
diff --git a/target/linux/qualcommax/image/ipq60xx.mk b/target/linux/qualcommax/image/ipq60xx.mk
index 307b7d3b5a..14147bcea5 100644
--- a/target/linux/qualcommax/image/ipq60xx.mk
+++ b/target/linux/qualcommax/image/ipq60xx.mk
@@ -55,10 +55,9 @@ define Device/jdc_ax1800-pro
 	DEVICE_PACKAGES := ipq-wifi-jdc_ax1800-pro kmod-fs-ext4 mkf2fs f2fsck kmod-fs-f2fs
 	BLOCKSIZE := 64k
 	KERNEL_SIZE := 6144k
-	IMAGES += kernel.bin rootfs.bin factory.bin
-	IMAGE/kernel.bin := append-kernel
-	IMAGE/rootfs.bin := append-rootfs | pad-rootfs | pad-to $$(BLOCKSIZE)
-	IMAGE/factory.bin := append-kernel | pad-to $$$${KERNEL_SIZE}  |  append-rootfs | append-metadata
+	IMAGES += factory.bin sysupgrade.bin
+	IMAGE/factory.bin := append-kernel | pad-to $$$${KERNEL_SIZE} | append-rootfs | pad-rootfs
+	IMAGE/sysupgrade.bin/squashfs := append-rootfs | pad-to $$$${BLOCKSIZE} | sysupgrade-tar rootfs=$$$$@ | append-metadata
 endef
 TARGET_DEVICES += jdc_ax1800-pro
 
diff --git a/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/01_leds b/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/01_leds
index 8cb9359a93..2bc060f834 100644
--- a/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/01_leds
+++ b/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/01_leds
@@ -8,8 +8,10 @@ board=$(board_name)
 
 case "$board" in
 jdc,ax1800-pro)
-    ucidef_set_led_netdev "wan" "WAN" "net_blue" "eth3"
-    ;;
+	ucidef_set_led_default "green" "GREEN" "green:status" "1"
+	ucidef_set_led_default "blue" "BLUE" "blue:status" "0"
+	ucidef_set_led_default "red" "RED" "red:status" "0"
+	;;
 yuncore,fap650)
 	ucidef_set_led_netdev "wlan5ghz" "WLAN 5GHz LED" "blue:wlan-5ghz" "wlan0" "tx rx"
 	ucidef_set_led_netdev "wlan2ghz" "WLAN 2.4GHz LED" "green:wlan-2ghz" "wlan1" "tx rx"
diff --git a/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network b/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network
index 62e531d395..999810d16c 100644
--- a/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network
+++ b/target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network
@@ -21,7 +21,7 @@ ipq60xx_setup_interfaces()
 		ucidef_set_interfaces_lan_wan "eth1 eth2" "eth0"
 		;;
 	jdc,ax1800-pro)
-		ucidef_set_interfaces_lan_wan "eth0 eth1 eth2" "eth3"
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
 		;;
 	netgear,wax214)
 		ucidef_set_interfaces_lan_wan "lan"
