diff --git a/modules/luci-base/root/sbin/soc-status b/modules/luci-base/root/sbin/soc-status
new file mode 100755
index 0000000..2430126
--- /dev/null
+++ b/modules/luci-base/root/sbin/soc-status
@@ -0,0 +1,78 @@
+#!/bin/sh
+# shellcheck disable=SC2155
+
+. /etc/openwrt_release
+
+CPUINFO_PATH="/proc/cpuinfo"
+CPUFREQ_PATH="/sys/devices/system/cpu/cpufreq"
+THERMAL_PATH="/sys/class/thermal"
+
+get_cpu_cores() {
+  local value="$(grep -c "processor" "$CPUINFO_PATH")"
+  [ -n "$value" ] || value="unknown"
+  echo "$value"
+}
+
+get_cpu_cur_freq() {
+  local value="$(awk '{printf("%.fMHz", $0 / 1000)}' "$CPUFREQ_PATH/policy0/cpuinfo_cur_freq")"
+  [ -n "$value" ] || value="0"
+  echo "$value"
+}
+
+get_cpu_temp() {
+  local value="$(awk '{ printf("%.1f°C", $0 / 1000) }' /sys/class/thermal/thermal_zone0/temp)"
+  [ -n "$value" ] || value="-1"
+  echo "$value"
+}
+
+get_5G_wifi_temp() {
+  local value="$(awk '{ printf("%.1f°C", $0 / 1000) }' /sys/class/ieee80211/phy0/hwmon0/temp1_input)"
+  [ -n "$value" ] || value="-1"
+  echo "$value"
+}
+
+get_2_4G_wifi_temp() {
+  local value="$(awk '{ printf("%.1f°C", $0 / 1000) }' /sys/class/ieee80211/phy1/hwmon1/temp1_input)"
+  [ -n "$value" ] || value="-1"
+  echo "$value"
+}
+
+get_cpu_usage() {
+  local value="$(top -b -n1 | awk '/^CPU/ { printf "%.0f%%\n", 100-$8 }')"
+  [ -n "$value" ] || value="-1"
+  echo "$value"
+}
+
+get_nss_usage() {
+  local value="$(grep '%' /sys/kernel/debug/qca-nss-drv/stats/cpu_load_ubi | awk '{print $2}')"
+  [ -n "$value" ] || value="-1"
+  echo "$value"
+}
+
+case "$1" in
+  cpu_cores)
+    get_cpu_cores
+    ;;
+  cpu_cur_freq)
+    get_cpu_cur_freq
+    ;;
+  cpu_temp)
+    get_cpu_temp
+    ;;
+  2_4G_wifi_temp)
+    get_2_4G_wifi_temp
+    ;;
+  5G_wifi_temp)
+    get_5G_wifi_temp
+    ;;
+  cpu_usage)
+    get_cpu_usage
+    ;;
+  nss_usage)
+    get_nss_usage
+    ;;
+  *)
+    echo "Usage: $0 {cpu_cores|cpu_cur_freq|cpu_temp|5G_wifi_temp|2_4G_wifi_temp|cpu_usage|nss_usage}"
+    exit 1
+    ;;
+esac
diff --git a/modules/luci-base/root/usr/share/rpcd/ucode/luci b/modules/luci-base/root/usr/share/rpcd/ucode/luci
index 3c4fea4..b5e3de6 100644
--- a/modules/luci-base/root/usr/share/rpcd/ucode/luci
+++ b/modules/luci-base/root/usr/share/rpcd/ucode/luci
@@ -581,6 +581,61 @@ const methods = {
 
 			return { result: ports };
 		}
+	},
+
+	getCoreInfo: {
+		call: function() {
+			let fd;
+			let result = {};
+
+			fd = popen('soc-status cpu_cores');
+			result.cores = trim(fd.read('all'));
+			fd.close();
+
+			fd = popen('soc-status cpu_cur_freq');
+			result.freq = trim(fd.read('all'));
+			fd.close();
+
+			return result;
+		}
+	},
+
+	getCoreTemp: {
+		call: function() {
+			let fd;
+			let result = {};
+
+			fd = popen('soc-status cpu_temp');
+			result.cpu = trim(fd.read('all'));
+			fd.close();
+
+			fd = popen('soc-status 5G_wifi_temp');
+			result.wifi_5G = trim(fd.read('all'));
+			fd.close();
+
+			fd = popen('soc-status 2_4G_wifi_temp');
+			result.wifi_2_4G = trim(fd.read('all'));
+			fd.close();
+
+			return result;
+		}
+	},
+
+	getCoreUsage: {
+		call: function() {
+			let fd;
+			let result = {};
+
+			fd = popen('soc-status cpu_usage');
+			result.cpu = trim(fd.read('all'));
+			fd.close();
+
+			fd = popen('soc-status nss_usage');
+			result.nss = trim(fd.read('all'));
+			fd.close();
+
+			return result;
+		}
 	}
 };
 
diff --git a/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js b/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
index 45f7b4a..ab115da 100644
--- a/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
+++ b/modules/luci-mod-status/htdocs/luci-static/resources/view/status/include/10_system.js
@@ -8,6 +8,21 @@ var callLuciVersion = rpc.declare({
 	method: 'getVersion'
 });
 
+var callCoreInfo = rpc.declare({
+	object: 'luci',
+	method: 'getCoreInfo'
+});
+
+var callCoreTemp = rpc.declare({
+	object: 'luci',
+	method: 'getCoreTemp'
+});
+
+var callCoreUsage = rpc.declare({
+	object: 'luci',
+	method: 'getCoreUsage'
+});
+
 var callSystemBoard = rpc.declare({
 	object: 'system',
 	method: 'board'
@@ -25,14 +40,20 @@ return baseclass.extend({
 		return Promise.all([
 			L.resolveDefault(callSystemBoard(), {}),
 			L.resolveDefault(callSystemInfo(), {}),
-			L.resolveDefault(callLuciVersion(), { revision: _('unknown version'), branch: 'LuCI' })
+			L.resolveDefault(callLuciVersion(), { revision: _('unknown version'), branch: 'LuCI' }),
+			L.resolveDefault(callCoreInfo(), {}),
+			L.resolveDefault(callCoreTemp(), {}),
+			L.resolveDefault(callCoreUsage(), {})
 		]);
 	},
 
 	render: function(data) {
 		var boardinfo   = data[0],
 		    systeminfo  = data[1],
-		    luciversion = data[2];
+		    luciversion = data[2],
+		    coreinfo    = data[3],
+		    coretemp    = data[4],
+		    coreusage   = data[5];
 
 		luciversion = luciversion.branch + ' ' + luciversion.revision;
 
@@ -54,7 +75,7 @@ return baseclass.extend({
 		var fields = [
 			_('Hostname'),         boardinfo.hostname,
 			_('Model'),            boardinfo.model,
-			_('Architecture'),     boardinfo.system,
+			_('Architecture'),     boardinfo.system + ' x ' + coreinfo.cores + ' (' + coreinfo.freq + ', CPU:' + coretemp.cpu + ', 2.4G:' + coretemp.wifi_2_4G + ', 5G:' + coretemp.wifi_5G + ')',
 			_('Target Platform'),  (L.isObject(boardinfo.release) ? boardinfo.release.target : ''),
 			_('Firmware Version'), (L.isObject(boardinfo.release) ? boardinfo.release.description + ' / ' : '') + (luciversion || ''),
 			_('Kernel Version'),   boardinfo.kernel,
@@ -64,7 +85,8 @@ return baseclass.extend({
 				systeminfo.load[0] / 65535.0,
 				systeminfo.load[1] / 65535.0,
 				systeminfo.load[2] / 65535.0
-			) : null
+			) : null,
+			_('使用率'),            'CPU:' + coreusage.cpu + ' ' + 'NSS:' + coreusage.nss
 		];
 
 		var table = E('table', { 'class': 'table' });
diff --git a/modules/luci-mod-status/root/usr/share/rpcd/acl.d/luci-mod-status.json b/modules/luci-mod-status/root/usr/share/rpcd/acl.d/luci-mod-status.json
index 45dd7d7..8f60a1e 100644
--- a/modules/luci-mod-status/root/usr/share/rpcd/acl.d/luci-mod-status.json
+++ b/modules/luci-mod-status/root/usr/share/rpcd/acl.d/luci-mod-status.json
@@ -3,7 +3,7 @@
 		"description": "Grant access to realtime statistics",
 		"read": {
 			"ubus": {
-				"luci": [ "getConntrackList", "getRealtimeStats" ],
+				"luci": [ "getConntrackList", "getRealtimeStats", "getCoreInfo", "getCoreTemp", "getCoreUsage" ],
 				"network.rrdns": [ "lookup" ]
 			}
 		}
